File: Bukkit-WolfLinkRPC/src/main/java/org/wolflink/paper/wolflinkrpc/analyse/BroadcastTextMessage.java
package org.wolflink.paper.wolflinkrpc.analyse;

import org.bukkit.Bukkit;
import org.jetbrains.annotations.NotNull;
import org.wolflink.common.wolflinkrpc.api.annotations.RemoteCallHandler;
import org.wolflink.common.wolflinkrpc.api.enums.ClientType;
import org.wolflink.common.wolflinkrpc.api.enums.DataPackType;
import org.wolflink.common.wolflinkrpc.api.interfaces.analyse.IAction;
import org.wolflink.common.wolflinkrpc.api.interfaces.analyse.IPredicate;
import org.wolflink.common.wolflinkrpc.api.interfaces.analyse.IRemoteHandler;
import org.wolflink.common.wolflinkrpc.entity.role.RPCUser;
import org.wolflink.paper.wolflinkrpc.App;

@RemoteCallHandler
public class BroadcastTextMessage implements IRemoteHandler {
    @NotNull
    @Override
    public IPredicate getPredicate() {
        return (rpcDataPack) -> rpcDataPack.getType() == DataPackType.TEXT_MESSAGE;
    }

    @NotNull
    @Override
    public IAction getAction() {
        return (rpcDataPack) -> {
            RPCUser sender = rpcDataPack.getSender();
            if(sender.getClientType() != ClientType.ANONYMOUS) Bukkit.broadcastMessage(App.RPC_CONFIGURATION.getProjectChineseName(true)+" §a"+sender.getUserName()+" §8» §f"+rpcDataPack.getJsonObject().get("msg").getAsString());
            else Bukkit.broadcastMessage(rpcDataPack.getJsonObject().get("msg").getAsString());
        };
    }
}

File: Mirai-WolfLinkRPC/src/main/kotlin/handler/remote/BroadcastTextMessage.kt
package org.wolflink.mirai.wolflinkrpc.handler.remote

import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch
import org.wolflink.common.wolflinkrpc.api.annotations.RemoteCallHandler
import org.wolflink.common.wolflinkrpc.api.enums.ClientType
import org.wolflink.common.wolflinkrpc.api.enums.DataPackType
import org.wolflink.common.wolflinkrpc.api.interfaces.analyse.IAction
import org.wolflink.common.wolflinkrpc.api.interfaces.analyse.IPredicate
import org.wolflink.common.wolflinkrpc.api.interfaces.analyse.IRemoteHandler
import org.wolflink.common.wolflinkrpc.entity.RPCDataPack
import org.wolflink.mirai.wolflinkrpc.App
import org.wolflink.mirai.wolflinkrpc.PersistenceConfig

@RemoteCallHandler
class BroadcastTextMessage : IRemoteHandler {
    override fun getAction(): IAction {
        return object : IAction{
            override fun invoke(datapack: RPCDataPack) {
                for (groupID in PersistenceConfig.enabledGroups)
                {
                    GlobalScope.launch {
                        val sender = datapack.sender
                        val msg = datapack.jsonObject.get("msg").asString
                        if(sender.clientType != ClientType.ANONYMOUS) App.bot.getGroup(groupID)?.sendMessage("[${datapack.sender.queueName}|${sender.userName}] $msg")
                        else App.bot.getGroup(groupID)?.sendMessage(msg)
                    }
                }
            }
        }
    }
    override fun getPredicate(): IPredicate {
        return object : IPredicate{
            override fun invoke(datapack: RPCDataPack): Boolean = (datapack.type == DataPackType.TEXT_MESSAGE && App.enabled)
        }
    }
}